indico/modules/events/abstracts/controllers/reviewing.py

```diff
@@ -183,10 +185,45 @@ class RHSubmitAbstractComment(RHAbstractBase):
     def _check_abstract_protection(self):
         return self.abstract.can_comment(session.user)
 
+    def _get_recipients(self):
+        judges = set()
+        conveners = set()
+        reviewers = set()
+
+        for entry in self.abstract.event.acl_entries:
+            if self.abstract.can_judge(entry.user):
+                judges.add(entry.user)
+            elif self.abstract.can_convene(entry.user):
+                conveners.add(entry.user)
+            elif self.abstract.can_review(entry.user):
+                reviewers.add(entry.user)
+        # Prevent sending duplicate emails for different roles
+        conveners -= judges
+        reviewers -= (judges | conveners)
+        return judges, conveners, reviewers
+
+    def _send_notification(self, recipients, comment):
+        for recipient in recipients:
+            tpl = get_template_module('events/abstracts/emails/comment.html', event=self.event, abstract=self.abstract,
+                                      submitter=session.user, comment=comment, recipient=recipient)
+            email = make_email(to_list=recipient.email, template=tpl, html=True)
+            send_email(email, self.event, 'Abstracts', session.user)
+
     def _process(self):
         form = AbstractCommentForm(abstract=self.abstract, user=session.user)
         if form.validate_on_submit():
             create_abstract_comment(self.abstract, form.data)
+            notify_on_new_comments = abstracts_reviewing_settings.get(self.event, 'notify_on_new_comments')
+            if notify_on_new_comments:
+                judges, conveners, reviewers = self._get_recipients()
+                comment = form.text.data
+                visibility = form.visibility.data
```

Comment:
This doesn't exist if you are commenting as the abstract submitter/author (if "Allow contributors in comments" is enabled):

```py
Traceback (most recent call last):
  File "/home/adrian/dev/indico/py3/env/lib/python3.9/site-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/adrian/dev/indico/py3/env/lib/python3.9/site-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/home/adrian/dev/indico/py3/src/indico/web/flask/util.py", line 79, in wrapper
    return obj().process()
  File "/home/adrian/dev/indico/py3/src/indico/web/rh.py", line 281, in process
    res = self._do_process()
  File "/home/adrian/dev/indico/py3/src/indico/modules/events/controllers/base.py", line 90, in _do_process
    return RHEventBase._do_process(self)
  File "/home/adrian/dev/indico/py3/src/indico/web/rh.py", line 252, in _do_process
    rv = self._process()
  File "/home/adrian/dev/indico/py3/src/indico/modules/events/abstracts/controllers/reviewing.py", line 220, in _process
    visibility = form.visibility.data
AttributeError: 'NoneType' object has no attribute 'data'
```

Reply:
Also, when a comment is posted with "Visible to contributors, reviewers, conveners, and judges", I think we should also notify:

- the submitter of the abstract
- anyone who previously commented on it (that way authors etc who commented will get notified, but we don't spam all other (co-)authors)